src/core/src/lib.rs

```diff
@@ -398,10 +398,22 @@ pub struct QueueSubmit<'a, B: Backend + 'a> {
 /// CommandBuffers will be later submitted to command queues instead of the device.
 pub trait CommandQueue<B: Backend> {
     /// Submit command buffers to queue for execution.
-    unsafe fn submit(&mut self, submit_infos: &[QueueSubmit<B>], fence: Option<&mut <B::Resources as Resources>::Fence>);
+    /// `fence` will be signalled after submission and _must_ be unsignalled.
+    // TODO: `access` legacy (handle API)
+    #[doc(hidden)]
+    unsafe fn submit(&mut self, submit_infos: &[QueueSubmit<B>], fence: Option<&mut <B::Resources as Resources>::Fence>,
```

Comment:
I marked it as `unsafe` as it's not supposed to be called. You could for example submit `GraphicsCommandBuffers` to a `ComputeQueue`, which will probably result in some driver error.